### PR TITLE
Created function getEvents for MailgunService.php

### PR DESCRIPTION
**Pull request by mariojrrc**
Function getLog from Mailgun service is deprecated. Created getEvents to replace it.

### CODE REVIEW

File: src/SlmMail/Service/MailgunService.php [hunk 1]
```diff
@@ -200,6 +200,52 @@ public function getLogs($limit = 100, $offset = 0)
...
+     *
+     * @link   http://documentation.mailgun.com/api-events.html
+     * @param  \Datetime $begin
+     * @param  \Datetime|null $end
+     * @param  int $limit
+     * @param  boolean $ascending
+     * @param array $fields
+     * @return array
+     */
+    public function getEvents(\DateTime $begin, \DateTime $end = null, $ascending = true, $limit = 300, array $fields = [])
```
Comment (bakura10) [inline src/SlmMail/Service/MailgunService.php]: Can you please import DateTime ?

**Comment by bakura10** [comment]
That looks like a solid change :). Can you please just do the small CS fix? I'll try to merge tomorrow ;). If you could also add a line in the changelog (let's set this as 2.1.1) for the change that woudl help as well.

File: src/SlmMail/Service/MailgunService.php (hunk 1 above)
Comment (mariojrrc) [inline src/SlmMail/Service/MailgunService.php]: Sure :)

**Comment by bakura10** [comment]
Let me know once everything's done :).

**Comment by mariojrrc** [comment]
I think I corrected everything. Let me know if you want some more change. 
I'll wait for the merge tomorrow.

Thanks

File: src/SlmMail/Service/MailgunService.php [hunk 2]
```diff
@@ -99,10 +100,10 @@ public function send(Message $message)
         }
 
         $parameters = array(
-            'from'    => $from->rewind()->toString(),
+            'from' => $from->rewind()->toString(),
             'subject' => $message->getSubject(),
-            'text'    => $this->extractText($message),
-            'html'    => $this->extractHtml($message)
+            'text' => $this->extractText($message),
```
Comment (bakura10) [inline src/SlmMail/Service/MailgunService.php]: Please revert those.

File: src/SlmMail/Service/MailgunService.php [hunk 3]
```diff
@@ -180,6 +181,72 @@ public function send(Message $message)
     }
 
     /**
+     * @param string $uri
+     * @param array $parameters
+     * @param bool $perDomain
+     * @return \Zend\Http\Client
+     */
+    private function prepareHttpClient($uri, array $parameters = array(), $perDomain = true)
```
Comment (bakura10) [inline src/SlmMail/Service/MailgunService.php]: What this all part of your change?

Comment (mariojrrc) [inline src/SlmMail/Service/MailgunService.php]: this all was done by cs-fix that rearranged the code.

Comment (bakura10) [inline src/SlmMail/Service/MailgunService.php]: Woké. I'm going to merge :).

**Comment by bakura10** [comment]
Done. Tagged as 2.1.1.

Thanks for your contribution!

**Comment by mariojrrc** [comment]
thanks for the fast reply :)

**Pull request merged**